test/util/matcher/matcher.go

```diff
@@ -60,3 +61,14 @@ func (m *AsyncOperationDocument) Matches(x interface{}) bool {
 func (m *AsyncOperationDocument) String() string {
 	return fmt.Sprintf("is equal to %v without comparing IDs", (*api.AsyncOperationDocument)(m))
 }
+
+// AssertErrHasWantMsg asserts that err.Error() is equal to wantMsg.
+func AssertErrHasWantMsg(t *testing.T, err error, wantMsg string) {
+	if err == nil && wantMsg != "" {
+		t.Errorf("did not get an error, but wanted error '%v'", wantMsg)
+	}
+
+	if err != nil && err.Error() != wantMsg {
+		t.Errorf("got error '%v', but wanted error '%v'", err, wantMsg)
+	}
+}
```

Comment:
this function doesn't really belong in here. You should probably put it in `test/util/error` (create the new package)

also the name is reading a little funny. Wouldn't be `AssertErrorMessage` be a bit nicer?

Reply:
Done.

Reply:
awesome thanks !

Reply:
Np. Thanks!